src/System.Windows.Forms/src/System/Windows/Forms/DragEventArgs.cs

```diff
@@ -60,5 +80,57 @@ public class DragEventArgs : EventArgs
         ///  Gets or sets which drag-and-drop operations are allowed by the target of the drag event.
         /// </summary>
         public DragDropEffects Effect { get; set; }
+
+        /// <summary>
+        ///  Gets or sets the drop description image type.
+        /// </summary>
+        public DropImageType DropImageType { get; set; }
+
+        /// <summary>
+        ///  Gets or sets the drop description text such as "Move to %1".
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// UI coloring is applied to the text in <see cref="MessageReplacementToken"/> if used by specifying %1 in <see cref="Message"/>.
+        /// </para>
+        /// </remarks>
+        public string? Message { get; set; }
+
+        /// <summary>
+        ///  Gets or sets the drop description text such as "Documents" when %1 is specified in <see cref="Message"/>.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// UI coloring is applied to the text in <see cref="MessageReplacementToken"/> if used by specifying %1 in <see cref="Message"/>.
+        /// </para>
+        /// </remarks>
+        public string? MessageReplacementToken { get; set; }
+
+        internal DragEventArgs Clone()
+        {
+            return (DragEventArgs)MemberwiseClone();
+        }
+
+        internal bool Equals(DragEventArgs? dragEventArgs)
+        {
+            if (dragEventArgs == this)
+            {
+                return true;
+            }
+
+            return dragEventArgs is not null
+                && dragEventArgs.Data is not null
+                && dragEventArgs.Data.Equals(Data)
+                && dragEventArgs.KeyState == KeyState
+                && dragEventArgs.X == X
+                && dragEventArgs.Y == Y
+                && dragEventArgs.AllowedEffect == AllowedEffect
+                && dragEventArgs.Effect == Effect
+                && dragEventArgs.DropImageType == DropImageType
+                && dragEventArgs.Message is not null
+                && dragEventArgs.Message.Equals(Message)
```

Comment:
What if `dragEventArgs.Message` is null and `Message` isn't?
```suggestion
                && dragEventArgs.Message == Message
```